src/modelfree/transparent.py

```diff
@@ -0,0 +1,188 @@
+    def __init__(self, sess, ob_space, ac_space, n_env, n_steps, n_batch, transparent_params,
+                 hiddens=None, scope="input", reuse=False, normalize=False):
+        LSTMPolicy.__init__(self, sess, ob_space, ac_space, n_env, n_steps, n_batch, hiddens,
+                            scope, reuse, normalize)
+        TransparentPolicy.__init__(self, transparent_params)
+
+    def get_obs_sizes(self):
+        ff_size = sum(self.hiddens[:-1])
+        return self.ob_space.shape[0], ff_size, ff_size, self.hiddens[-1]
+
+    def step(self, obs, state=None, mask=None, deterministic=False):
+        action = self.deterministic_action if deterministic else self.action
+        outputs = [action, self.value_flat, self.state_out, self.neglogp, self.ff_out]
+        feed_dict = self._make_feed_dict(obs, state, mask)
+        a, v, s, neglogp, ff = self.sess.run(outputs, feed_dict)
+        state = []
+        for x in s:
+            state.append(x.c)
+            state.append(x.h)
+        state = np.array(state)
+        state = np.transpose(state, (1, 0, 2))
+
+        # right now 'hid' only supports hidden state of policy
+        # which is the last of the four state vectors
+        transparent_objs = (obs, np.concatenate(ff['policy']),
+                            np.concatenate(ff['value']), state[:, -1, :])
+        transparency_dict = self.get_transparency_dict(transparent_objs)
+        return a, v, state, neglogp, transparency_dict
+
+
+class TransparentMlpPolicyValue(TransparentPolicy, MlpPolicyValue):
+    """gym_compete MlpPolicyValue which is also transparent."""
+    def __init__(self, sess, ob_space, ac_space, n_env, n_steps, n_batch, transparent_params,
+                 hiddens=None, scope="input", reuse=False, normalize=False):
+        MlpPolicyValue.__init__(self, sess, ob_space, ac_space, n_env, n_steps, n_batch,
+                                hiddens=hiddens, scope=scope, reuse=reuse, normalize=normalize)
+        TransparentPolicy.__init__(self, transparent_params)
+
+    def get_obs_sizes(self):
+        return self.ob_space.shape[0], sum(self.hiddens), sum(self.hiddens), None
+
+    def step(self, obs, state=None, mask=None, deterministic=False):
+        action = self.deterministic_action if deterministic else self.action
+        outputs = [action, self.value_flat, self.neglogp, self.ff_out]
+        a, v, neglogp, ff = self.sess.run(outputs, {self.obs_ph: obs})
+
+        transparent_objs = (obs, np.concatenate(ff['policy']), np.concatenate(ff['value']), None)
+        transparency_dict = self.get_transparency_dict(transparent_objs)
+        return a, v, self.initial_state, neglogp, transparency_dict
+
+
+class TransparentCurryVecEnv(CurryVecEnv):
+    """CurryVecEnv that provides transparency data about its policy in its infos dicts."""
+    def __init__(self, venv, policy, agent_idx=0):
+        super().__init__(venv, policy, agent_idx)
+        self.underlying_policy = policy.policy
```

Comment:
I assume the `policy` is actually a `BaseRLModel`? Perhaps call it `model` instead then?

Reply:
Actually this seems to be an issue with the base `CurryVecEnv`, I think we should change both to `model` if we're passing in `BaseRLModel`.

Reply:
It seems like this naming convention actually goes deeper, if I'm understanding properly. CurryVecEnv is used in EmbedVictimWrapper, where the `policy` argument of CurryVecEnv is supplied by the `victim` argument of EmbedVictimWrapper. If you go up another level to maybe_embed_victim in train.py, that victim object is returned by a `load_policy` function. I think I lean slightly towards keeping `policy` rather than `model`, just because the former is a more informative term and adds to code readability, but either way, it seems like we'll need to do a consistency sweep on that change more broadly throughout the codebase.